src/test/java/streams/part1/exercise/Exercise2.java

```diff
@@ -0,0 +1,119 @@
+package streams.part1.exercise;
+
+import lambda.data.Employee;
+import lambda.data.JobHistoryEntry;
+import lambda.data.Person;
+import org.junit.Test;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.Comparator;
+import java.util.List;
+
+import static org.junit.Assert.assertEquals;
+
+@SuppressWarnings({"ConstantConditions", "unused"})
+public class Exercise2 {
+
+    @Test
+    public void calcAverageAgeOfEmployees() {
+        List<Employee> employees = getEmployees();
+
+        Double expected = employees.stream().
+                mapToInt(employee -> employee.getPerson().getAge()).
+                average().
+                getAsDouble();
+
+        assertEquals(33.66, expected, 0.1);
+    }
+
+    @Test
+    public void findPersonWithLongestFullName() {
+        List<Employee> employees = getEmployees();
+
+        Person expected = employees.stream().
+                map(Employee::getPerson).
+                max(Comparator.comparingInt(person -> person.getFullName().length())).
+                get();
+
+        assertEquals(expected, employees.get(1).getPerson());
+    }
+
+    @Test
+    public void findEmployeeWithMaximumDurationAtOnePosition() {
+        List<Employee> employees = getEmployees();
+
+        Employee expected = employees.stream().
+                max(Comparator.comparingInt(employee -> employee.getJobHistory().
+                        stream().
+                        mapToInt(JobHistoryEntry::getDuration).
+                        max().
+                        getAsInt())).
+                get();
+
+        assertEquals(expected, employees.get(4));
+    }
+
+    /**
+     * Вычислить общую сумму заработной платы для сотрудников.
+     * Базовая ставка каждого сотрудника составляет 75_000.
+     * Если на текущей позиции (последняя в списке) он работает больше трех лет - ставка увеличивается на 20%
+     */
+    @Test
+    public void calcTotalSalaryWithCoefficientWorkExperience() {
+        List<Employee> employees = getEmployees();
+
+        int expected = employees.stream().
+                map(Employee::getJobHistory).
+                mapToInt(jobHistoryEntries -> jobHistoryEntries.
+                        get(jobHistoryEntries.size() - 1).
+                        getDuration() > 3 ?
+                        90_000 : 75_000).
```

Comment:
It's bad practice to use "Magic numbers". In addition, lines 68-71 are not easy to read.

Reply:
fixed 5b2b6d2c99001016f2caeac17c6d8dcda372cf57